### ISSUE TITLE
Scope for variable read/write access.

### ISSUE THREAD

**Issue by JacksonKearl** [atom/language-java#221]
## Summary

It would be helpful to add a scope to the textmate grammar denoting variable read/write access. For example in typescript, the scope `variable.other.readwrite` is applied. 

## Motivation

Currently the only scope given to variable read/write access is `source.java`, which means users and themes cannot configure the appearance of these accesses in themes. 

## Additional context

<img width="349" alt="Screen Shot 2020-01-19 at 4 12 55 PM" src="https://user-images.githubusercontent.com/8586769/72690987-8fe84e00-3ad6-11ea-95ce-7ffe3732a750.png">

https://www.reddit.com/r/vscode/comments/equyio/im_using_the_horizon_theme_in_vscode_why_do_local/

**Comment by sadikovi** [comment]
No, we don't do that currently. Can you give examples when it is useful?

**Comment by JacksonKearl** [comment]
Currently the only scope given to variable read/write access is source.java, which means users and themes cannot configure the appearance of variable uses in themes.

Many themes attempt to control variable colors, and they should be able to work in Java in addition to other languages.

**Comment by sadikovi** [comment]
Okay, makes sense. I am a bit snowed under at the moment due to bunch of other open-source work, would appreciate a pull request :).
